src/server/worker/common/common.go

```diff
@@ -49,6 +50,29 @@ func HashDatum(pipelineName string, pipelineSalt string, inputs []*Input) string
 	return client.DatumTagPrefix(pipelineSalt) + hex.EncodeToString(hash.Sum(nil))
 }
 
+// DatumIDV2 computes the ID of a datum.
+// TODO: This needs more discussion.
+func DatumIDV2(inputs []*Input) string {
+	var files []string
+	for _, input := range inputs {
+		files = append(files, input.Name+strings.ReplaceAll(input.FileInfo.File.Path, "/", "_"))
+	}
+	return strings.Join(files, "-")
+}
+
+// HashDatumV2 computes the hash of a datum.
+func HashDatumV2(pipelineName string, pipelineSalt string, inputs []*Input) string {
+	hash := sha256.New()
```

Comment:
Is it worth coordinating the hash function with PFS.  Maybe moving it to a separate package?

Reply:
Yeah, that probably makes sense. If a user configures a different hash function than the default then it probably makes sense to use that for pfs hashing as well as computing the datum hashes. I'll go ahead and change it to setup a pfs hasher, then we can change it when decide how we want to expose the configuration for the pfs hash.